uxarray/_scrip.py

```diff
@@ -1,11 +1,123 @@
-from warnings import warn
+import xarray as xr
+import numpy as np
 
 
-def _read_scrip(filepath):
-    """This is just a function placeholder.
+def _to_ugrid(in_ds, out_ds):
+    """If input dataset (ds) file is an unstructured SCRIP file, function will
+    reassign SCRIP variables to UGRID conventions in output file (outfile).
 
-    Not implemented.
+    Parameters
+    ----------
+    in_ds : :class:`xarray.Dataset`
+        Original scrip dataset of interest being used
+
+    out_ds : :class:`xarray.Variable`
+        file to be returned by _populate_scrip_data, used as an empty placeholder file
+        to store reassigned SCRIP variables in UGRID conventions
     """
-    warn("Function not implemented yet. FYI, attempted to read SCRIP file: " +
-         str(filepath))
-    # TODO: create ds
+
+    if in_ds['grid_area'].all():
+
+        # Create Mesh2_node_x/y variables from grid_corner_lat/lon
+        # Turn latitude scrip array into 1D instead of 2D
+        corner_lat_xr = in_ds['grid_corner_lat']
+        corner_lat = corner_lat_xr.values.ravel()
```

Comment:
Redundant memory usage through the initialization of `corner_lat_xr` (and similarly `corner_lon_xr` below) looks like avoidable. Those variables are only actually used for their `shape`s, which is for example (384,4) in this dataset, and it could be read through some other variables coming from SCRIP datasets. If you can do that, you can get rid of `corner_lat_xr` and `corner_lon_xr` and initialize `corner_lat ` and `corner_lat` as:

`corner_lat = in_ds['grid_corner_lat'].values.ravel()`

Reply:
@erogluorhan I used the built in PyCharm profiler on the tests for the function and it appears I reduced the memory usage with the most recent push/changes. Please feel free to double check those stats on your end too!